Add eslint rule

chanha-park opened the pull request:
나날이 strict 해지는 eslint입니다. 추가된 내용입니다

1. no-unused-expressions: side effect 없고 리턴값도 안쓰이는 expression을 잡아줍니다. 예시: `const a = 1; a + 1;`

2. no-unnecessary-qualifier: 불필요한 qualifier를 잡아줍니다.
```
enum A {
  B,
  C = A.B,
}
```
3. no-unnecessary-condition: 아래 예시같이 애초에 false가 될 수 없는 인자는 검사가 필요없습니다.
```
function foo(arg: 'bar' | 'baz') {
  // arg is never nullable or empty string, so this is unnecessary
  if (arg) {
  }
}
```
4. no-unnecessary-boolean-literal-compare: `boolean` 타입인 값의 경우 `if (x === true)` 말고 `if (x)`, `if (!x)` 같은 식으로 쓰도록 합니다.

5. no-non-null-asserted-nullish-coalescing: `foo! ?? bar` 같은거 잡아줍니다. null 아니라고 느낌표 붙여놓고선 ?? 쓰면 이상하잖아요.

6. no-warning-comments, no-console: todo, fixme, xxx같은 주석과 console log 같은 코드들 경고해줍니다.

this PR close #92 
이렇게 하면 main에 머지될때 태그된 이슈가 자동으로 닫힙니다 아마..?

chanha-park committed:
add eslint rule

chanha-park committed:
add eslint rule:
no-console, no-warning-comments를 warning option으로 추가

TODO, FIXME 등의 주석, console.log 등을 쓰는 코드를 경고해줍니다

chanha-park commented:
마음에 안든다 / 너무 과하다 / 취향에 안맞는다 하는 부분은 자유롭게 의견 주세요

hdoo42 reviewed (approved):
eslint룰이 정말 좋은게 많네요 ㅋㅋ

chanha-park committed:
add eslint rule: import, export consistency rules.

key, member, imports order warning

[1,2,3].sort() function require compare function

chanha-park committed:
add eslint rule: template literal without expressions

chanha-park committed:
add eslint rule: eqeqeq

chanha-park commented:
일관된 import, export형식에 대한 규칙, == 대신 ===을 쓰는 규칙, 일반 문자열은 백틱 대신 따옴표로 쓰도록 하는 규칙이 추가되었습니다.
import 순서는 좀 성가셔서 제외하셔도..?

chanha-park committed:
sort, ordering 관련 룰 다시 제거
strict-boolean-expressions option을 더욱 strict하게

chanha-park committed:
Merge branch 'websocket-hooks' into add-eslint-rule
